### PR TITLE
Support cloud storage sink

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12151
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-csv.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+- `commit-ts` of the last row in a single CSV file must be greater than that in the previous row.
+- The same transaction of a single table cannot be stored in different CSV files.
+- Multiple tables of the same transaction can be stored in different CSV files.
+
+## Definition of the data format
+
+Each column in a row of the CSV file is defined as follows:
+
+- Column 1: The operation-type indicator, including `I`, `D`, and `U`. `I` means `INSERT`, `U` means `UPDATE`, and `D` means `DELETE`.
```
Comment (hfxsd) [inline ticdc/ticdc-csv.md:43]:
```suggestion
- Column 1: The operation-type indicator, including `I`,  `U`, and `D`. `I` means `INSERT`, `U` means `UPDATE`, and `D` means `DELETE`.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+### Configure NFS in sink URI
+
+The following configuration saves row change events to NFS:
+
+```
+shell
+--sink-uri="file:///my-directory/prefix"
+```
+
+Other parameters optional in the URI are as follows:
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:48]:
```suggestion
Other optional parameters in the URI are as follows:
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+| Parameter         | Description                                             |
+| :------------ | :------------------------------------------------ |
+| `worker-count` | Concurrency for saving data changes to cloud storage in the downstream (optional, default value: `16`, value range: [`1`, `512`] |
+| `flush-interval` | Interval for saving data changes to cloud storage in the downstream (optional, default value: `5s`, value range: [`2s`, `10m`] |
+| `file-size` | A data change file is stored to cloud storage if its bytes exceed the value of this parameter (optional, default value: `67108864`, value range: [`67108864`, `536870912`]) |
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:54]: 中文的取值范围下限是 1048576
```suggestion
| `file-size` | A data change file is stored to cloud storage if its bytes exceed the value of this parameter (optional, default value: `67108864`, value range: [`1048576`, `536870912`]) |
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+```
+
+- `protocol`: specifies the data transmission protocol, or the storage type, for example, `s3://xxxxx`.
+- `prefix`: specifies the user-defined parent directory, for example, `s3://bucket/bbb/ccc`.
+- `schema`: specifies the schema name, for example, `s3://bucket/bbb/ccc/test`.
+- `table`: specifies the table name, for example, `s3://bucket/bbb/ccc/test/table1`.
+- `table-version-separator`: specifies the separator that separates the file path by the table version, for example, `s3://bucket/bbb/ccc/test/table1/9999`.
+- `partition-separator`: specifies the separator that separates the file path by the table partition, for example, `s3://bucket/bbb/ccc/test/table1/9999/20`
+- `date-separator`: classifies the files by the transaction commit date. Value options are:
+    - none: no date-separator. For example, all files with `test.table1` version being `9999` are saved to `s3://bucket/bbb/ccc/test/table1/9999`.
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:79]:
```suggestion
    - `none`: no date-separator. For example, all files with `test.table1` version being `9999` are saved to `s3://bucket/bbb/ccc/test/table1/9999`.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    - `year`: the separator is the year a transaction is committed, for example, `s3://bucket/bbb/ccc/test/table1/9999/2022`.
+    - `month`: the separator is the year and month a transaction is committed, for example,`s3://bucket/bbb/ccc/test/table1/9999/2022-01`.
+    - `day`: the separator is the year, month, and day a transaction is committed, for example,`s3://bucket/bbb/ccc/test/table1/9999/2022-01-02`.
+- `num`: saves the serial number of the file that records the data change, for example, `s3://bucket/bbb/ccc/test/table1/9999/2022-01-02/CDC000005.csv`.
+- `extension`: specifies the extension of the file. TiDB v6.5.0 supports the CSV format only.
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:84]: 中文版说支持两种格式。
```suggestion
- `extension`: specifies the extension of the file. TiDB v6.5.0 supports the CSV and canal-json formats only.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    - `ColumnLength`: Column length. For details, see [Data type](#data-type).
+    - `ColumnPrecision`: Column precision. For details, see [Data type](#data-type).
+    - `ColumnScale`: The number of digits following the decimal point (the scale). For details, see [Data type](#data-type).
+    - `ColumnNullable`: The column can be null when the value of this option is `true`.
+    - `ColumnIsPk`: The column is part of the primary key when the value of this option is `true`.
+- `TableColumnsTotal`: The size of the `TableColumns` array.
+
+### Data type
+
+Data type is defined as `T(M[, D])`. For details, see [Data Types](/data-type-overview.md).
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:167]:
```suggestion
The data type is defined as `T(M[, D])`. For details, see [Data Types](/data-type-overview.md).
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+```shell
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{ST}",
+    "ColumnLength":"{M}"
+}
+```
+
+#### Enum and Set types
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:252]: 代码示例中用的是 ENUM/SET，保持一致吧。
```suggestion
#### ENUM/SET types
```

Comment (shichun-0415) [reply ticdc/ticdc-sink-to-cloud-storage.md:252]: 但其实是两种类型，见 [`ENUM` type](https://docs.pingcap.com/tidb/dev/data-type-string#enum-type) 和 [`SET` type](https://docs.pingcap.com/tidb/dev/data-type-string#set-type)，我改成大写然后 and 连接吧，鉴于文档中不推荐用 / 表示并列关系。

File: ticdc/ticdc-changefeed-config.md [hunk 8]
```diff
@@ -69,7 +69,7 @@ enable-old-value = true
 
 # Filter rules.
 # Filter syntax: <https://docs.pingcap.com/tidb/stable/table-filter#syntax>.
-rules = ['*.*', '!test.*']
+rules = ['*.*When the downstream is a storage service, the protocol can only be set to canal-json or csv.', '!test.*']
```
Comment (zhaoxinyu) [inline ticdc/ticdc-changefeed-config.md:72]: Is this a typo?

File: ticdc/ticdc-changefeed-config.md [hunk 9]
```diff
@@ -100,7 +100,27 @@ dispatchers = [
...
+date-separator = 'none'
+# Whether to use partitions as the separation string. The default value is false, which means that partitions in a table are not stored in separate directories. For details, see <https://docs.pingcap.com/zh/tidb/dev/ticdc-sink-to-cloud-storage#data-change-records)>
+enable-partition-separator = false
+
+# Since v6.5.0, TiCDC supports saving data changes to storage services in CSV format. Ignore the following configurations if you replicate data to MQ and MySQL sinks.
+[sink.csv]
+# Delimiter between fields. The value must be an ASCII character and defaults to `,`.
+delimiter = ','
+# The quotation that surrounds the fields. The default value is `"`, which means no quotation is used.
+quote = '"'
```
Comment (zhaoxinyu) [inline ticdc/ticdc-changefeed-config.md:121]: This is incorrect. This translation is not consistent with the Chinese version. The default value `"` means double quote will be used.

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+--sink-uri="file:///my-directory/prefix"
+```
+
+Other optional parameters in the URI are as follows:
+
+| Parameter         | Description                                             |
+| :------------ | :------------------------------------------------ |
+| `worker-count` | Concurrency for saving data changes to cloud storage in the downstream (optional, default value: `16`, value range: [`1`, `512`] |
+| `flush-interval` | Interval for saving data changes to cloud storage in the downstream (optional, default value: `5s`, value range: [`2s`, `10m`] |
+| `file-size` | A data change file is stored to cloud storage if its bytes exceed the value of this parameter (optional, default value: `67108864`, value range: [`1048576`, `536870912`]) |
```
Comment (zhaoxinyu) [inline ticdc/ticdc-sink-to-cloud-storage.md:61]:
```suggestion
| `file-size` | A data change file is stored to cloud storage if the number of bytes exceed the value of this parameter (optional, default value: `67108864`, value range: [`1048576`, `536870912`]) |
```

File: ticdc/ticdc-changefeed-config.md (hunk 8 above)
Comment (shichun-0415) [inline ticdc/ticdc-changefeed-config.md:72]:
```suggestion
rules = ['*.*', '!test.*']
```

File: ticdc/ticdc-changefeed-config.md [hunk 11]
```diff
@@ -100,7 +100,27 @@ dispatchers = [
...
+# Date separator type of the file directory. Value options are `none`, `year`, `month`, and `day`. `none` is the default value and means that the date is not separated. For details, see <https://docs.pingcap.com/zh/tidb/dev/ticdc-sink-to-cloud-storage#data-change-records>.
+date-separator = 'none'
+# Whether to use partitions as the separation string. The default value is false, which means that partitions in a table are not stored in separate directories. For details, see <https://docs.pingcap.com/zh/tidb/dev/ticdc-sink-to-cloud-storage#data-change-records)>
+enable-partition-separator = false
+
+# Since v6.5.0, TiCDC supports saving data changes to storage services in CSV format. Ignore the following configurations if you replicate data to MQ and MySQL sinks.
+[sink.csv]
+# Delimiter between fields. The value must be an ASCII character and defaults to `,`.
+delimiter = ','
+# The quotation that surrounds the fields. The default value is `"`, which means no quotation is used.
```
Comment (shichun-0415) [inline ticdc/ticdc-changefeed-config.md:120]:
```suggestion
# The quotation that surrounds the fields. The default value is `"`. When the value is empty, no quotation is used.
```

File: ticdc/ticdc-changefeed-config.md (hunk 9 above)
Comment (shichun-0415) [reply ticdc/ticdc-changefeed-config.md:121]: Fixed in [68fb1b4](https://github.com/pingcap/docs/pull/10998/commits/68fb1b4b4fa1b6d9d27d1b9c0250d3d5a3b5eefc)

File: ticdc/ticdc-changefeed-config.md (hunk 8 above)
Comment (shichun-0415) [reply ticdc/ticdc-changefeed-config.md]: Fixed in [68fb1b4](https://github.com/pingcap/docs/pull/10998/commits/68fb1b4b4fa1b6d9d27d1b9c0250d3d5a3b5eefc)

**Review by zhaoxinyu — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 12]
```diff
@@ -0,0 +1,279 @@
...
+    - `month`: the separator is the year and month a transaction is committed, for example, <code>s3://bucket/bbb/ccc/test/table1/9999/**2022-01**</code>.
+    - `day`: the separator is the year, month, and day a transaction is committed, for example, <code>s3://bucket/bbb/ccc/test/table1/9999/**2022-01-02**</code>.
+- `num`: saves the serial number of the file that records the data change, for example, <code>s3://bucket/bbb/ccc/test/table1/9999/2022-01-02/CDC**000005**.csv</code>.
+- `extension`: specifies the extension of the file. TiDB v6.5.0 supports the CSV and canal-json formats.
+
+> **Note:**
+>
+> The table version changes in the following two cases:
+>
+> - After DDL operations are performed, the table version is the TSO when the DDL is executed in the upstream TiDB. However, the change of the table version does not mean the change of the table schema. For example, adding a comment to a column does not cause the `schema.json` file content to change.
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:97]:
```suggestion
> - After a DDL operation is performed, the table version is the TSO when the DDL is executed in the upstream TiDB. However, the change of the table version does not mean the change of the table schema. For example, adding a comment to a column does not cause the `schema.json` file content to change.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 13]
```diff
@@ -0,0 +1,279 @@
...
+{
+    "checkpoint-ts":433305438660591626
+}
+```
+
+- `checkpoint-ts`: Transactions with `commit-ts` smaller than `checkpoint-ts` are written to the target storage in the downstream.
+
+### DDL events
+
+When DDL events cause the table version to change, TiCDC switches to the new path to write data change records. For example, when the version of `test.table1` changes from `9999` to `10000`, data will be written to the path `s3://bucket/bbb/ccc/test/table1/10000/2022-01-02/CDC000001.csv`. In addition, when DDL events occur, TiCDC generates a `schema.json` file to save the table schema information.
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:120]:
```suggestion
When DDL events cause the table version to change, TiCDC switches to a new path to write data change records. For example, when the version of `test.table1` changes from `9999` to `10000`, data will be written to the path `s3://bucket/bbb/ccc/test/table1/10000/2022-01-02/CDC000001.csv`. In addition, when DDL events occur, TiCDC generates a `schema.json` file to save the table schema information.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 14]
```diff
@@ -0,0 +1,279 @@
...
+
+Table  schema information is saved in the following path:
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:122]:
```suggestion
Table schema information is saved in the following path:
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+
+Metadata is saved in the following path:
+
+```shell
+{protocol}://{prefix}/metadata
+```
+
+Metadata is a JSON-formatted file, for example:
+
+```shell
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:110]:
```suggestion
```json
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+- `Version`: Protocol version of the storage sink.
+- `TableVersion`: Table version.
+- `Query`：DDL statement.
+- `TableColumns`: An array of one or more maps, each of which describes a column in the source table.
+    - `ColumnName`: Column name.
+    - `ColumnType`: Column type. For details, see [Data type](#data-type).
+    - `ColumnLength`: Column length. For details, see [Data type](#data-type).
+    - `ColumnPrecision`: Column precision. For details, see [Data type](#data-type).
+    - `ColumnScale`: The number of digits following the decimal point (the scale). For details, see [Data type](#data-type).
+    - `ColumnNullable`: The column can be null when the value of this option is `true`.
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:179]:
```suggestion
    - `ColumnNullable`: The column can be NULL when the value of this option is `true`.
```

Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:179]:
```suggestion
    - `ColumnNullable`: The column can be NULL when the value of this option is `true`.
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+
+## Configure sink URI
+
+This section describes how to configure storage services in the changefeed URI, including Amazon S3, Azure Blob Storage, and NFS.
+
+### Configure S3 or Azure Blob Storage in sink URI
+
+The URI parameters of Amazon S3 and Azure Blob Storage in TiCDC are the same as their URL parameters in BR. For details, see [Backup storage URL format](/br/backup-and-restore-storages.md#url-format-description).
+
+### Configure NFS in sink URI
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:48]:
```suggestion
### Configure NFS
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+- `--sink-uri`: The downstream address of the changefeed. For details, see [Configure sink URI](#configure-sink-uri).
+- `--start-ts`: The starting TSO of the changefeed. TiCDC starts pulling data from this TSO. The default value is the current time.
+- `--target-ts`: The ending TSO of the changefeed. TiCDC stops pulling data until this TSO. The default value is empty, which means that TiCDC does not automatically stop pulling data.
+- `--config`: The configuration file of the changefeed. For details, see [TiCDC changefeed configuration parameters](/ticdc/ticdc-changefeed-config.md).
+
+## Configure sink URI
+
+This section describes how to configure storage services in the changefeed URI, including Amazon S3, Azure Blob Storage, and NFS.
+
+### Configure S3 or Azure Blob Storage in sink URI
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:44]:
```suggestion
### Configure Amazon S3 or Azure Blob Storage
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+{
+    "ColumnName":"COL1",
+    "ColumnType":"{ST}",
+    "ColumnLength":"{M}"
+}
+```
+
+#### Enum and Set types
+
+The `ENUM` and `SET` types are defined as follows in `schema.json`:
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:276]:
```suggestion
The Enum and Set types are defined as follows in `schema.json`:
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 20]
```diff
@@ -0,0 +1,283 @@
+---
+title: Replicate Data to Storage Services
+summary: Learn how to replicate data to storage services using TiCDC, and learn about the storage path of the replicated data.
+---
+
+# Replicate Data to Storage Services
+
+> **Warning:**
+>
+> This feature is experimental. It is not recommended to use it in the production environment.
```
[the remark this answers is not in the document]
Comment (shichun-0415) [reply ticdc/ticdc-sink-to-cloud-storage.md:10]: IMO, the original is more frequently used for experimental features, and `the current version` may make it hard to maintain.

**Pull request merged**